### ISSUE TITLE
Variogram function inaccessible

### ISSUE THREAD

**Issue by dhruvbalwada** [GliderToolsCommunity/GliderTools#10]
**Describe the bug**
Currently the variogram function is inaccessible. 

**To Reproduce**
```
import glidertools as gt
filenames = '../tests/data/p54203*.nc'
gt.load.seaglider_show_variables(filenames)
names = ['ctd_depth','ctd_time','salinity']
ds_dict = gt.load.seaglider_basestation_netCDFs(filenames,
                        names, return_merged=True, keep_global_attrs=False)
#
depth = ds_dict['sg_data_point'].ctd_depth
dives = ds_dict['sg_data_point'].dives
var   = ds_dict['sg_data_point'].salinity
m = (depth<150) & (dives > 306) & (dives < 314)
vargram = gt.mapping.variogram(var, dist, depth, dives, mask=m)
```

**Expected behavior**
A variogram object should be passed to `vargram`.

**Screenshots**
Instead the following error is obtained
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-18-c759fb535bd0> in <module>
     10 var   = ds_dict['sg_data_point'].salinity
     11 m = (depth<150) & (dives > 306) & (dives < 314)
---> 12 vargram = gt.mapping.variogram(var, dist, depth, dives, mask=m)

AttributeError: module 'glidertools.mapping' has no attribute 'variogram'

**Additional context**
Python version 3.7.6

**Comment by luke-gregor** [comment]
Have you installed `pykrige`? Adding this to the `requirements.txt` file caused the installation to fail, so I've made it an optional manual installation. If this package is available, you should be able to access the variogram function. 

I think there's quite a bit of room for improvement with the variogram function. Tried to automate the x/y scaling, but it's not very transparent - makes it tough to understand what's going on.

**Comment by dhruvbalwada** [comment]
Yeah, I have pykrige installed. I think the problem is because the `variogram` function inside a `try`, for some reason it is not being detected. Maybe a problem with python version?

**Comment by luke-gregor** [comment]
🤔  I can't create the problem with my setup. And importing pykrige as standalone works fine?

**Comment by dhruvbalwada** [comment]
Yes, importing pykrige directly works. Might be a problem with my setup, I will dig into it more. Thanks for taking a look.

**Comment by dhruvbalwada** [comment]
I dug into this matter a little more, because I ran into a similar problem again. This time with the function `gt.plot.section3D`. I installed GliderTools in a new environment, directly from pip this time. 
I get the exact same error in this case too `AttributeError: type object 'plot_functions' has no attribute 'section3D'` . 

Note that both functions `variogram` and `section3D` sit inside `if` statements, inside some class. In both cases you are testing if a certain module is installed. 
Also note that in both cases the if statement is not even called, I mean that the error does not read something like ''install "plotly" for 3D plotting sections'. 

I do not have enough familiarity with the OOP and scope structures of Python, but maybe you have a better sense of what is going on. I think it is a problem of scope based on reading: https://stackoverflow.com/questions/54630146/how-to-access-the-variables-after-if-condition-when-the-variable-is-defined-insi 

Is it possible that the version of Python you are using has a different definition of scopes than mine? I am using 3.7
